zerver/decorator.py

```diff
@@ -502,6 +505,15 @@ def _wrapped_view_func(request, *args, **kwargs):
             if param.func_var_name in kwargs:
                 continue
 
+            if param.argument_type == 'body':
+                from zerver.lib.response import json_error
+                try:
+                    val = ujson.loads(request.body)
+                except ValueError:
+                    return json_error("Malformed JSON")
```

Comment:
I think this should probably raise an exception, rather than returning, based on the other stuff in this file.